prompt/prompt-structure/src/commonMain/kotlin/ai/koog/prompt/structure/json/JsonStructuredData.kt

```diff
@@ -67,26 +66,29 @@ public class JsonStructuredData<TStruct>(
         // TODO: Class.simpleName is the only reason to make the function inline, perhaps we can hide most of the implementation
         /**
          * Factory method to create JSON structure with auto-generated JSON schema.
+         *
+         * @param id Unique identifier for the structure.
+         * @param serializer Serializer used for converting the data to and from JSON.
+         * @param json JSON configuration instance used for serialization.
+         * @param schemaFormat Format of the generated schema, can be simple or detailed.
+         * @param maxDepth Maximum recursion depth when generating schema to prevent infinite recursion for circular references.
+         * @param descriptions Optional map of serial class names and property names to descriptions.
+         * If a property/type is already described with [LLMDescription] annotation, value from the map will override this description.
```

Comment:
Let's also please add good example in KDoc about how to form the descriptions map.
I mean, `"className.propertyName" to "property description"`

Reply:
Also, maybe let's name it `descriptionOverrides` if it's an overriding map?

Reply:
Regarding testing edge cases, that's what I already modified in tests, to cover these cases you described. I didn't mark each field with annotation though because it looks redundant, and it's also nice to test that description is added only for these fields where specified it, and is skipped for other fields.

Regarding examples and parameter rename: sounds good, will do it